### PR TITLE
fix(ui) Filter search through schemas based on non-editable tags and terms as well

### PR DESCRIPTION
**Pull request by chriscollins3456**
Previously we were only filtering schemas by matching on editable tags and terms. This adds an additional check to see if we match non-editable tags and terms on the fields themselves. This involved a little refactoring of my check matching methods.

## Checklist
- [ ] The PR conforms to DataHub's [Contributing Guideline](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [ ] Links to related issues (if applicable)
- [ ] Tests for the changes have been added/updated (if applicable)
- [ ] Docs related to the changes have been added/updated (if applicable). If a new feature has been added a Usage Guide has been added for the same.
- [ ] For any breaking change/potential downtime/deprecation/big changes an entry has been made in [Updating DataHub](https://github.com/datahub-project/datahub/blob/master/docs/how/updating-datahub.md)

### CODE REVIEW

File: datahub-web-react/src/app/entity/shared/tabs/Dataset/Schema/utils/filterSchemaRows.ts [hunk 1]
```diff
@@ -53,12 +51,20 @@ export function filterSchemaRows(
     const expandedRowsFromFilter = new Set();
 
     rows.forEach((row) => {
-        if (shouldInclude(row.fieldPath, row.fieldPath, formattedFilterText, filteredFieldPathsByMetadata)) {
+        if (
+            matchesFieldName(row.fieldPath, formattedFilterText) ||
+            matchesEditableTagsOrTerms(row, filteredFieldPathsByEditableMetadata) ||
+            tagsOrTermsMatchFilter(row, formattedFilterText, entityRegistry) // non-editable tags and terms
```
Comment (jjoyce0510) [inline datahub-web-react/src/app/entity/shared/tabs/Dataset/Schema/utils/filterSchemaRows.ts:57]: Should this be matchesTagsOrTerms?

Comment (jjoyce0510) [reply datahub-web-react/src/app/entity/shared/tabs/Dataset/Schema/utils/filterSchemaRows.ts:57]: (For consistency)

Comment (chriscollins3456) [reply datahub-web-react/src/app/entity/shared/tabs/Dataset/Schema/utils/filterSchemaRows.ts:57]: certainly

**Review by jjoyce0510 — APPROVED** [review]
LGTM - Thank you!

**Pull request merged**